### PR TITLE
Fix for composite report HIDs.

### PR DESCRIPTION
**Pull request by Franticware**
There was an extraneous byte in boot mouse report for composite device type mice. In existing code, this was only handled in process_generic_report, not tuh_hid_report_received_cb. This is a fix for that.
I also tested it on pi pico and it works fine.

### CODE REVIEW

File: usb/host/host_cdc_msc_hid/hid_app.c [hunk 1]
```diff
@@ -99,17 +99,26 @@ void tuh_hid_umount_cb(uint8_t dev_addr, uint8_t instance)
 void tuh_hid_report_received_cb(uint8_t dev_addr, uint8_t instance, uint8_t const* report, uint16_t len)
 {
   uint8_t const itf_protocol = tuh_hid_interface_protocol(dev_addr, instance);
+  uint8_t const rpt_count = tuh_hid_instance_count(dev_addr);
```
Comment (lurch) [inline usb/host/host_cdc_msc_hid/hid_app.c:102]: _Tiny_ nitpick, but perhaps `report_count` would be a better variable name? When I first read this I thought that `rpt_count` meant "repeat count" :wink:

**Comment by lurch** [comment]
ping @hathach just in case a similar problem affects TinyUSB too?

File: usb/host/host_cdc_msc_hid/hid_app.c (hunk 1 above)
Comment (Franticware) [reply usb/host/host_cdc_msc_hid/hid_app.c]: Done.

**Comment by Franticware** [comment]
> ping @hathach just in case a similar problem affects TinyUSB too?

Maybe, is he able to see this PR because he is tagged?

**Comment by lurch** [comment]
Sorry, that was a question aimed at the TinyUSB developer, not at yourself Vojtěch :wink: 
(i.e. I wasn't saying "Should we ping Ha Thach?", I was saying "I'm pinging Ha Thach here (by mentioning his GitHub username), just in case TinyUSB is also affected".)

**Comment by Franticware** [comment]
@lurch Should I do something to move forward the approval/merge process?

**Comment by lurch** [comment]
@Franticware Nope, I think the only thing to do at this point is be patient... :slightly_smiling_face:

**Comment by Franticware** [comment]
I now realize this fix is not good enough. Unfortunately there are other issues (the data passed to tuh_hid_report_received_cb callback by tinyusb is incomplete) and I am not able to fix it at this point.

**Comment by Franticware** [comment]
The key to solving this is using the following as part of TinyUSB initialization:
tuh_hid_set_default_protocol(HID_PROTOCOL_REPORT);

Then the report data must be interpreted according to the descriptor.

The default HID_PROTOCOL_BOOT does not work very well with many USB mice that I have tested so far.

**Comment by lurch** [comment]
> The key to solving this is using the following as part of TinyUSB initialization: tuh_hid_set_default_protocol(HID_PROTOCOL_REPORT);
> 
> Then the report data must be interpreted according to the descriptor.
> 
> The default HID_PROTOCOL_BOOT does not work very well with many USB mice that I have tested so far.

Would you like to create a new issue and/or PR? I suspect that commenting on a closed PR will mean that your comment will quickly get forgotten / lost.

**Pull request closed without merging**